map/map.mbt

```diff
@@ -35,6 +35,37 @@ enum Map[K, V] {
   Tree(K, V, Int, Map[K, V], Map[K, V])
 }
 
+type Pair[K, V] (K, V) derive(Debug, Show)
+
+pub fn to_string[K : Debug + Show, V : Debug + Show](
+  self : Map[K, V]
+) -> String {
+  self.to_vec_map(fn { k, v => Pair::Pair((k, v)) }).to_string()
+}
+
+
+test "to_string" {
+  let buf = Buffer::make(100)
+  fn write_buf(x : Show){
+    buf.write_string(x.to_string())
+    buf.
+  }
+  
+  Map::[(1, "hell")]|> write_buf
+  buf.expect(
```

Comment:
@lijunchen note currently buf lack some method, eg, reset, and expect message is not very friendly for grepping